src/components/header/Header.tsx

```diff
@@ -69,6 +80,11 @@ export const Header: React.FC = () => {
             iconName="magnifying-glass"
             isActive={searchIsActive}
           />
+          {totalQuantity > 0 ? (
+            <Badge className="badge-icon" label={totalQuantity} />
```

Comment:
Rather than just having this floating in the header detached from everything let's make the IconButton render children then we can pass this into it to improve the hierarchy